src/decorators/Service.ts

```diff
@@ -93,10 +96,15 @@ export function Service<T, K extends keyof T>(optionsOrServiceName?: ServiceOpti
     } else {
         return function(target: Function) {
 
+
+            Object.defineProperty(target, "originalName", { value: target.name || target.constructor.name, writable: false });
+            Object.defineProperty(target, "name", { value: createHash("md5").update(`${target}`).digest("hex"), writable: true });
```

Comment:
You really shouldn't define a property on someone elses object like this it can lead to really subtle bugs for people who have intentionally used the name or originalName properties already. You could safely use a `Symbol()` here instead however.

Then later when resolving the key instead of just using the functions name, look it up by the symbol.